### PR TITLE
Finished piece of cake ex

### PR DESCRIPTION
**Pull request by michall1900**
Hey itayday and talbarfm,
Please check the exercise,
Thanks in advance .

### CODE REVIEW

File: 5.2/Piece of cake.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+"""
+Calculate price for recipe's ingredients
+"""
+
+
+def get_recipe_price(prices: dict, optionals: list = None, **gram_ingredients: dict) -> float or None:
+    """Receives dictionary when key its ingredients name's and item is their price for 100 g,
```
Comment (talbarfm) [inline 5.2/Piece of cake.py]: newline after """

File: 5.2/Piece of cake.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+"""
+Calculate price for recipe's ingredients
+"""
+
+
+def get_recipe_price(prices: dict, optionals: list = None, **gram_ingredients: dict) -> float or None:
```
Comment (talbarfm) [inline 5.2/Piece of cake.py]: float or None is incorrect, read about typing.Union

File: 5.2/Piece of cake.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    """Receives dictionary when key its ingredients name's and item is their price for 100 g,
+    list of optional ingredients that someone don't want to buy at all and dictionary with same keys as the first
+    but it's items is how many grams do we need to buy.
+    If keys are not the same at both of dictionaries, return null. Otherwise, calculate and return the final price.
+    :param prices: Keys are ingredients names, items are their price for 100 gram
+    :param optionals: List of items (as keys) that we don't want to buy.
+    :param gram_ingredients: Keys as prices dictionary and items as grams we need to buy.
+    :return: none when there is error, else, total price when
+    """
+    if not prices.keys() == gram_ingredients.keys():
```
Comment (talbarfm) [inline 5.2/Piece of cake.py]: Cant you supply a price that will not be used in this function? seems like you are too harsh in your limitations, without a reason

File: 5.2/Piece of cake.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        print("There are missing key/s at one of the dicts")
```
Comment (talbarfm) [inline 5.2/Piece of cake.py]: I would raise an exception and not print anything - afterwards I can catch it and do anything I want with it

File: 5.2/Piece of cake.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return None
+    final_price = 0
```
Comment (talbarfm) [inline 5.2/Piece of cake.py]: Read about functools.reduce and replace this for loop with it

File: 5.2/Piece of cake.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    for key, item in prices.items():
+        if optionals is not None and key in optionals:
+            continue
+        final_price += item * gram_ingredients.get(key) / 100
+
+    return final_price
+
+
+def main_piece_of_cake():
+    # Just doing and printing some tests.
```
Comment (talbarfm) [inline 5.2/Piece of cake.py]: You dont have to document this, its obvious

File: 5.2/Piece of cake.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    print(get_recipe_price({'chocolate': 18, 'milk': 8}, chocolate=200, milk=100))
```
Comment (talbarfm) [inline 5.2/Piece of cake.py]: maybe a lambda to reduce the repeats on these two lines? just for the fun of it?

**Review by talbarfm — CHANGES_REQUESTED** [review]
Good job!!!@

**Review by lovodkin93 — APPROVED** [review]
very good!

**Pull request merged**